src/app/options/options.component.ts

```diff
@@ -14,8 +18,11 @@ import { SortOptions } from '../interfaces';
 })
 export class OptionsComponent implements OnInit {
 
-  showDOMSource: boolean = false;
+  links: LinkData[];
```

Comment:
I'm not sure if we want to store LinkData[] in this component the long run. I imagine option component responsibility should be fairly simple  - storing a simple configuration. in the component. Do you think it's possible for link service to holds the data instead?

Reply:
I think you could make a case that the "true" value of LinkList is held in the chromeLinkService, but in reality it actually holds the observable that pushes the latest version of linkList. Based on what I have learned looking into the best way to manage data for Angular, a better way to think about what observables do though is that they are just passing on values to the components who are the ones that are supposed to "hold" or use the data. Since the object is shared by reference, it's more apt to think of the components as holding a reference to the list of links than holding the actual data anyway. I considered not giving the options component access to the links directly, but it ended up leaving the code a lot clunkier, and more confusing in a myriad of ways:

1. It led to an excessive amount of callbacks between the options component and the link service that muddled what was actually going on. Specifically because we have to manage the asynchronous nature of asking chrome for the links as well as asking chrome about the storage. 
2. There is no really good reason why ChromeLinkService should be responsible for calling FetchService in the first place, before I had put it there because of a lack of a better alternative, and because it was always called immediately after receiving the links. However, now that the responsibility of coordinating when the links should be fetched falls to the options component (both with the button and the auto fetch option), I think it is much cleaner and more straightforward, to move that responsibility into the options component.
3. While having all of this logic would make the options component a bit simpler, it would add two responsibilities to ChromeLinkService that are unrelated to it's core purpose. In addition it would add a lot of confusion to the interaction between the options component and ChromeLinkService. Simply managing this interaction would take a lot of a space in the options component anyway. I think that having the options component simply take care of the logic that is triggered by the buttons in its purview is the most clear way to distinguish the responsibility of the components and services, as it leaves ChromeLinkService with a much more defined and easier to follow role. Otherwise ChromeLinkService starts becoming an accumulation of unrelated tasks. Additionally it is consistent with the way that we pass data throughout the application.

At the end of the day it felt like I was shoe-horning in a lot of unnecessary calls and logic just to force ChromeLinkService to do a bunch of work, that it probably shouldn't be doing anyway.  If anything I think we should consider moving the observables into the cross-component-data service since that is what we are now doing with the rest of the data.

Reply:
Thanks for well-thought response!

I see your point on not making ChromeLinkService holds the data. In general, I think your previous commits made things looks better. It is much easier to understand the responsibility of ChromeLinkService and CrossComponentDataService now. However, I'm still worried that Option component can be too bloated as a component.

As we discussed offline: Unfortunately, I don't have a perfect solution myself. Let's moving on with this and we can iterate how to handle things in Options component later.